src/main/java/org/zalando/nakadi/service/timeline/TimelineSyncImpl.java

```diff
@@ -0,0 +1,365 @@
+package org.zalando.nakadi.service.timeline;
+
+import java.io.Closeable;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.apache.curator.framework.recipes.locks.InterProcessLock;
+import org.apache.curator.framework.recipes.locks.InterProcessSemaphoreMutex;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.WatchedEvent;
+import org.apache.zookeeper.Watcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
+import org.zalando.nakadi.util.UUIDGenerator;
+
+@Service
+@Profile("!test")
+public class TimelineSyncImpl implements TimelineSync {
+    private static class DelayedChange {
+        private final int version;
+        private final Set<String> lockedEventTypes;
+
+        private DelayedChange(final int version, final Collection<String> lockedEventTypes) {
+            this.version = version;
+            this.lockedEventTypes = new HashSet<>(lockedEventTypes);
+        }
+
+        @Override
+        public String toString() {
+            return "{version=" + version + ", lockedEventTypes=" + lockedEventTypes + '}';
+        }
+    }
+
+    private static final String ROOT_PATH = "/timelines";
+    private static final Charset CHARSET = Charset.forName("UTF-8");
```

Comment:
In other places we use what is already there in guava
`com.google.common.base.Charsets.UTF_8`